heron/common/src/python/handler/access/heron.py

```diff
@@ -112,117 +114,161 @@ def get_topologies_states():
   request_url = create_url(TOPOLOGIES_URL_FMT) + "/states"
   raise tornado.gen.Return((yield fetch_url_as_json(request_url)))
 
+
+@tornado.gen.coroutine
+def _get_topologies(cluster, role=None, env=None):
+  endpoint = create_url(TOPOLOGIES_URL_FMT)
+  params = dict(cluster=cluster)
+  if role:
+    params['role'] = role
+  if env:
+    params['environ'] = env
+  request_url = tornado.httputil.url_concat(endpoint, params)
+  raise tornado.gen.Return((yield fetch_url_as_json(request_url)))
+
+
 ################################################################################
 # Get the list of topologies given a cluster
 ################################################################################
-@tornado.gen.coroutine
 def get_cluster_topologies(cluster):
-  endpoint = create_url(TOPOLOGIES_URL_FMT)
-  request_url = tornado.httputil.url_concat(endpoint, dict(cluster=cluster))
-  raise tornado.gen.Return((yield fetch_url_as_json(request_url)))
+  return _get_topologies(cluster)
+
+
+################################################################################
+# Get the list of topologies given a cluster submitted by a given role
+################################################################################
+def get_cluster_role_topologies(cluster, role):
+  return _get_topologies(cluster, role=role)
+
+
+################################################################################
+# Get the list of topologies given a cluster submitted by a given role under
+# a given environment
+################################################################################
+def get_cluster_role_env_topologies(cluster, role, env):
+  return _get_topologies(cluster, role=role, env=env)
+
 
 ################################################################################
 # Get the execution state of a topology in a cluster
 ################################################################################
 @tornado.gen.coroutine
-def get_execution_state(cluster, environ, topology):
+def get_execution_state(cluster, environ, topology, role=None):
+  params = dict(cluster=cluster, environ=environ, topology=topology)
+  if role:
```

Comment:
`if role` and `if role is not None` are different, can you check and see which one is better here?